### ISSUE TITLE
change default shell to fish

### ISSUE THREAD

**Issue by iaacornus** [iaacornus/Fedora-OSTree-Setup#14]
- [ ] install fish
- [ ] change default shell to fish

**Comment by trytomakeyouprivate** [comment]
This should be optional. What are the advantages for you?

**Comment by iaacornus** [comment]
> This should be optional. What are the advantages for you?

yes, it's optional, users can agree with the prompt and change it or not

**Comment by trytomakeyouprivate** [comment]
1. add fish to rpm-ostree command

2. add this to reboot command

```
cat ~/.bashrc >> ~/.config/fish/config.fish
echo "Enter /usr/bin/fish"
sudo lchsh $USER
```

I am curious, what doesnt run on fish? Any bash commands that are important?

**Comment by trytomakeyouprivate** [comment]
question: can lchsh be automated to use /usr/bin/fish as argument?

**Comment by trytomakeyouprivate** [comment]
maybe create a seperate fish appstarter

```
printf """[Desktop Entry]
Name=Fish Shell
Exec=fish
Terminal=true
Type=Application""" > ~/.local/share/applications/fish.desktop
```

**Comment by trytomakeyouprivate** [comment]
another question is how to remove this huge shell "username" (?):

```bash
user@fedora /v/h/u/.c/fish> 
```

**Comment by trytomakeyouprivate** [comment]
fish shell now needs another way to add the paths to $PATH

- ~/.cargo/bin
- ~/.local/bin
- ~/.bin

**Comment by trytomakeyouprivate** [comment]
you could also just change to fish via

```
echo "fish" >> ~/.bashrc
```

a more extreme way would be changing the shell permanently

```
usermod -s /usr/bin/fish $USER
```

**Comment by trytomakeyouprivate** [comment]
unsure. Some people said it doesnt do anything unexpected, some others say it breaks the system. I would just keep fish as the shell for the Terminal app, and use good old bash for everything else. Fish is pretty nice!

**Comment by trytomakeyouprivate** [comment]
PROBLEM:

Distrobox using Fedora doesnt work if fish is not installed. Some script detecting if a distrobox using fedora exists and installing fish there is needed.

**Comment by john-ghatas** [comment]
The root shell should not be modified, this should be optional anyway since shells are a user preference kinda thing.

**Comment by trytomakeyouprivate** [comment]
Yes agreed. Do you know if there are Gnome terminal profiles? On KDE I dont have to change any shell, just create a new profile using fish and set that as default.

The alternative would be to have an Appstarter opening fish and set that to Ctrl+Alt+T

**Comment by john-ghatas** [comment]
Yes gnome supports profiles

![image](https://user-images.githubusercontent.com/14125058/227779050-2523b06f-d445-42ad-a7c8-2e05a1c5dda9.png)

**Comment by trytomakeyouprivate** [comment]
nice! Would you like to create a config file we can automatically print somewhere, so that users just have to select the profile? Maybe even set the profile as default?

**Comment by john-ghatas** [comment]
Sure that would only be a custom command to invoke fish then right?

**Comment by john-ghatas** [comment]
I found the following link to dump and save Gnome terminal profiles

**Comment by john-ghatas** [comment]
Created the following config file 

```
> dconf dump /org/gnome/terminal/legacy/profiles:/
[/]
list=['b1dcc9dd-5262-4d8d-a863-c897e6d979b9']

[:b1dcc9dd-5262-4d8d-a863-c897e6d979b9]
use-custom-command=false

[legacy/profiles:]
list=['8844b46f-4827-4106-9afb-77fcf76521fa']

[legacy/profiles:/:8844b46f-4827-4106-9afb-77fcf76521fa]
custom-command='fish'
use-custom-command=true
visible-name='fish'
```

Then load that config
```

```

Used https://gist.github.com/fdaciuk/9ec4d8afc32063a6f74a21f8308e3807

**Comment by john-ghatas** [comment]
Managed to export the configuration but I can't find any working example on how to import it
